### ISSUE TITLE
Failed to handle notification textDocument/didSave

### ISSUE THREAD

**Issue by tomslee** [openlawlibrary/pygls#108]
I'm trying to develop with pygls using vim and the vim-lsc Language Server Client plugin (https://github.com/natebosch/vim-lsc) as a client. I get this error in pygls:

ERROR  Failed to handle notification textDocument/didSave: Object....
Traceback (most recent call last):
 File.... protocol.py line 272, in _get_handler
   return self.fm.buildin_features[feature_name]
KeyError: 'textDocument/didSave'

I've commented out the line in vim-lsc that sends this notification so I can carry on working, but I thought you may want to know about it.

**Comment by danixeee** [comment]
Hello @tomslee,

[Specification](https://microsoft.github.io/language-server-protocol/specifications/specification-current/#textDocument_didSave) says that client should send `textDocument/didSave` notification _only_ if server allowed this in its capabilities:

```
Server Capability:

property name (optional): textDocumentSync.save
property type: boolean | SaveOptions where SaveOptions is defined as follows:

...
```

Unless `didSave` is registered with `@serever.feature(types.TEXT_DOCUMENT_DID_SAVE)` you should not send this notification. AFAIK, this applies to most of the notifications that client sends to the server.

On the other hand, I think that `pygls` do have some issues with building the server capabilities response, so I will keep this issue open and you can left comments if you find more issues related to this.

**Comment by muffinmad** [comment]
@danixeee 

How server can allow `didSave` notification? Although `TextDocumentSyncOptions` is in the `pygls/types.py`, only `TextDocumentSyncKind` can be passed as `textDocumentSync` in server capabilities.

**Comment by tomslee** [comment]
Thanks for the update. I have not seen other problems. I'll notify the owner of the vim-lsc plugin about sending unhandled notifications.

**Comment by danixeee** [comment]
This should be solved by #139. Please reopen if that is not the case.
